src/dict.c

```diff
@@ -870,136 +869,163 @@ void dictTwoPhaseUnlinkFree(dict *d, dictEntry *he, dictEntry **plink, int table
     dictResumeRehashing(d);
 }
 
+#define DICT_SET_KEY(e, k)                                                                                             \
```

Comment:
Also, is this really needed as a macro? I thought only the values needed macros because they have the variadic underlying types.

Reply:
Yes, it needs to be a macro. Since we're writing the code in C, macros are the only way to approximate the type flexibility that C++ templates offer. This approach is essential for implementing a common accessor pattern that works with any structs sharing the field name "value," even when the "value" field is of different types. Inline functions do not provide the necessary type flexibility in this scenario, as they can't handle varying types for the same field name across different structs. While variadic functions can handle a variable number of arguments, they don't solve the problem of accessing struct fields with different types.

Reply:
I think we are talking about two different things. When the value is involved you are 100% correct, but when it's just the key there is only one type. This is related to an earlier comment from Viktor.